.github/workflows/preview-docs.yml

```diff
@@ -1,28 +1,88 @@
 name: Preview docs
 
 on:
-  pull_request_target:
+  workflow_run:
+    workflows: [docs]
+    types: [requested, completed]
 
 jobs:
+  placeholder_comment:
+    if: github.repository == 'mlflow/mlflow' && github.event.workflow_run.status == 'requested'
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write # preview_docs.py comments on PRs
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: ./.github/actions/setup-python
+      - name: Post placeholder comment
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.workflow_run.pull_requests[0].number }}
+        run: |
+          python dev/preview_docs.py \
+            --commit-sha ${{ github.event.workflow_run.head_sha }} \
+            --pull-number $PR_NUMBER \
+            --workflow-run-id ${{ github.run_id }} \
+            --docs-workflow-run-url ${{ github.event.workflow_run.html_url }} \
+            --stage requested
+
   main:
-    if: github.repository == 'mlflow/mlflow'
+    if: github.repository == 'mlflow/mlflow' && github.event.workflow_run.status == 'completed'
     runs-on: ubuntu-latest
     permissions:
-      statuses: read # preview_docs.py checks PR statuses
       pull-requests: write # preview_docs.py comments on PRs
     timeout-minutes: 10
     steps:
+      - name: Handle docs workflow failure
+        if: github.event.workflow_run.conclusion == 'failure'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR_NUMBER: ${{ github.event.workflow_run.pull_requests[0].number }}
+        run: |
+          python dev/preview_docs.py \
+            --commit-sha ${{ github.event.workflow_run.head_sha }} \
+            --pull-number $PR_NUMBER \
+            --workflow-run-id ${{ github.run_id }} \
+            --docs-workflow-run-url ${{ github.event.workflow_run.html_url }} \
+            --stage failed
+          exit 0 # Exit the job if the docs workflow failed
```

Comment:
do we need this line?

Reply:
I believe if we don't put it, it will fall through if `workflow_run.conclusion == 'failure'`. The rest of the job is handling the success case.

Reply:
I think early-exit is still unsupported: https://github.com/orgs/community/discussions/82744

Reply:
Separated the success case into another job with dependency on the failure check success.